### PR TITLE
chore: upgrade to next 12

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Upgrading to Next 12, but still keeping our babel in favor of SWC (as we don't have all the plugins in SWC yet)
- Does introduce a 5% bundle increate so it's a consideration to make. (Known issue if you don't go full SWC in Next 12)

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-226 #done

### CODE REVIEW

**Comment by idoshamun** [comment]
Can we perhaps migrate to SWC? What's the missing part?

**Comment by rebelchris** [comment]
@idoshamun I think at this point we use the following the webapp babel:

- macros -> Not available in SWC yet
- babel-plugin-module-resolver (For our monorepo dependencies so not sure we can use the native next module paths for this)
- react-remove-properties -> Available in SWC
- dynamic-import-node (Can possibly be removed

**Review by idoshamun — APPROVED** [review]
The only thing to check is if we bundle (p)react twice to the final bundle.
When I used babel and webpack I had to carefully adjust it so the shared package will not bundle one version of react and the webapp another.

**Comment by rebelchris** [comment]
@idoshamun great point, do you have any guidance on how you checked this before?
Not sure how to tell the difference for the final build.

**Comment by idoshamun** [comment]
You have to use the bundle inspector. We can do it together. Not sure I remember it well enough to guide you through it without seeing it firsthand

File: packages/shared/src/components/post/AuthorOnboarding.tsx [hunk 1]
```diff
@@ -39,10 +38,9 @@ export function AuthorOnboarding({
         <li>Gain reputation points by earning upvotes on articles you wrote</li>
       </ol>
       <div
-        className="grid grid-flow-col gap-x-4 mt-6"
+        className="grid grid-flow-col gap-x-4 mt-6 max-w-[4.625rem]"
```
Comment (sshanzel) [inline packages/shared/src/components/post/AuthorOnboarding.tsx:41]: I think the `sizeN` has a multiplier of `0.25`.
```suggestion
        className="grid grid-flow-col gap-x-4 mt-6 max-w-[18.5rem]"
```

File: packages/shared/src/components/utilities.tsx [hunk 2]
```diff
@@ -118,7 +118,7 @@ export const NewCommentContainer = classed(
 
 export const ResponsivePageContainer = classed(
   PageContainer,
-  'py-6 laptop:border-theme-divider-tertiary laptop:border-l laptop:border-r laptop:min-h-screen',
+  'py-6 laptop:border-theme-divider-tertiary laptop:border-l laptop:border-r laptop:min-h-screen pb-[1.4375rem]',
```
Comment (sshanzel) [inline packages/shared/src/components/utilities.tsx:121]: The original value was 23 which is 5.75rem, with that, I think we can round it off to just 24 to lessen arbitrary value. What do you think?
```suggestion
  'py-6 laptop:border-theme-divider-tertiary laptop:border-l laptop:border-r laptop:min-h-screen pb-24',
```

**Review by sshanzel — APPROVED** [review]
The performance gains here is the next level Dev Experience! Thank you for the initiative! ❤️ 

Just had some values for correction, but all looks so amazing to me! 🚢

**Review by kirillkurko — APPROVED** [review]

File: packages/shared/src/components/utilities.tsx (hunk 2 above)
Comment (rebelchris) [reply packages/shared/src/components/utilities.tsx:121]: Ah good point, took it very literal, but yeah think we can do this rounding.

**Comment by rebelchris** [comment]
@idoshamun I've run the bundle analyzer, not really sure what to look for, but it seems the shared doesn't import it's own react right?

Screenshot 2022-07-12 at 10 42 31
Screenshot 2022-07-12 at 10 43 34

**Comment by idoshamun** [comment]
Yes if you can't find preact twice. Or both react and preact in there then it should be fine

**Pull request merged**